index.js

```diff
@@ -9,9 +9,12 @@ const registerPlugin = require('./lib/registerPlugin');
 
 // Plugins depended on by the `meta` package.
 const corePlugins = new Map();
-Object.keys(meta.dependencies).forEach(
-  name => /^meta-/.test(name) && corePlugins.set(name, require.resolve(name))
-);
+Object.keys(meta.dependencies).forEach(name => {
+  if (/^meta-/.test(name)) {
+    const packagePath = require.resolve(path.join(name, 'package.json'));
```

Comment:
The `registerPlugin` function uses `path.basename` on the resolved plugin path, so we want to ensure something like `meta-git` resolves to `/path/to/meta-git` instead of `/path/to/meta-git/index.js`. We play it safe in case a core plugin decides to move its main module to a sub-directory.